src/View/Sidebar/NetworkRow.vala

```diff
@@ -20,16 +20,47 @@
  * Authors : Jeremy Wootten <jeremy@elementaryos.org>
  */
 
-public class Sidebar.NetworkRow : Sidebar.AbstractMountableRow {
-    public NetworkRow (
-        string name, string uri, Icon gicon, SidebarListInterface list, Mount? mount
-    ) {
+public class Sidebar.NetworkRow : Sidebar.VolumelessMountRow {
+    public NetworkRow (string name, string uri, Icon gicon, SidebarListInterface list,
+                               bool pinned, bool permanent,
+                               string? _uuid, Mount? _mount) {
+        Object (
+            custom_name: name,
+            uri: uri,
+            gicon: gicon,
+            list: list,
+            pinned: pinned,
+            permanent: permanent,
+            uuid: _uuid,
+            mount: _mount
+        );
 
-        base (
-            name, uri, gicon, list,
-            true, // Pinned,
-            mount == null || !mount.can_unmount (), // permanent if no mount or unmountable,
-            null, null, null, //No uuid, drive or volume
-            mount);
+        if (mount != null) {
+            string scheme, hostname;
+            try {
+                var connectable = NetworkAddress.parse_uri (uri, 0);
+                scheme = connectable.scheme;
+                hostname = connectable.hostname;
+            } catch (Error e) {
+                scheme = Uri.parse_scheme (uri);
```

Comment:
this may set `scheme` to `null`, which would lead to a weird custom name

Reply:
Fixed